### DISCUSSION TITLE
job filtering from worker perspective

### DISCUSSION THREAD

**Discussion by nebula2** [elsa-workflows/elsa-core#3481] [Q&A]
Hey :)

I currently evaluate to move my workflow stuff to elsa. Got one question  though.

Is it possible to add something to a job to let a worker decide if it is the right one to execute it (capabilities)?

Two examples:
1. I am releasing a new application version with changes to the implementation of an activity. The activity requires a worker to be on version 2.
I want to prevent workers on version 1 from taking jobs that require a higher version.

2. My workers have different specs. There are "small" machines that have only 2gb of ram and there are others that have 8gb of ram (for example).
Activities that are flagged with something custom like "high resource usage" should only be worked off by workers that have the same flag.

sorry if I misuse terminology - hope you understand what I mean.

**Comment by dpfei** [comment]
您好，我已收到您的邮件。看到后我会尽快回复。感谢您的关注。                              顺祝商祺！
